examples_utils/benchmarks/environment_utils.py

```diff
@@ -30,6 +45,26 @@ def get_mpinum(command: str) -> int:
     return mpinum
 
 
+def check_poprun_env_variables(benchmark_name: str, cmd: str):
+    """Check if poprun environment variables have been set prior to running.
+
+    Args:
+        benchmark_name (str): The name of the benchmark being run
+        cmd (str): The command being run
+
+    """
+
+    # Check if any of the poprun env vars are required but not set
+    for env_var, help_msg in POPRUN_VARS.items():
+        if f"${env_var}" in cmd and os.getenv(env_var) is None:
+            err = (f"Environment variable {env_var} is a value passed to an "
+                   f"argument in {benchmark_name}, but has not been set in "
+                   f"your environment.\nHint: {env_var} = {help_msg}")
+            logger.error(err)
+
+            raise EnvironmentError(err)
```

Comment:
I like the list comprehension but I feel makes things harder for devs and users?

For devs its a little less clear, and for users I think firing 5 hints at once in a huge error message might be overwhelming. It would be helpful if they dont need to run 5 times to realise they're missing 5 environment variables, but I feel the clarity of having one error per run is better? 

If you feel strongly about this, I can compromise on this change, its still pretty good, but otherwise I would prefer not to.

Reply:
I think splitting the list comprehension makes it clearer for a developer:

```suggestion
    missing_env_vars = [
        env_var for env_var in POPRUN_VARS.keys()
        if f"${env_var}" in cmd and os.getenv(env_var) is None
    ]
    if missing_env_vars:
        err = (
            f"{len(missing_env_vars)} environment variables are needed by command"
            f"{cmd_name} but are not defined: {missing_env_vars}. Hints: \n"
        ) + "".join([
            f"\n\t{missing} : {POPRUN_VARS[missing]}"
            for missing in missing_env_vars
        ])
        logger.error(err)
        raise EnvironmentError(err)
```

And then lets us make a nice error message which is not too overwhelming?

```
```

Reply:
I prefer that to having to rerun the code to get it to fail again.

Reply:
ok this is pretty nice, I agree. Didnt think of it like this earlier, commiting into this PR. thanks.